src/managed_data/client.py

```diff
@@ -54,6 +54,18 @@ def invoke(self, endpoint: str, data: dict):
 
         pass  # pragma: no cover
 
+    @abstractmethod
+    def download(self, endpoint: str, data: dict):
+        """abstract method to download a resource to disk"""
+
+        pass  # pragma: no cover
+
+    @abstractmethod
+    def put(self, endpoint: str, data: dict):
+        """abstract method to implement a PUT request"""
+
+        pass  # pragma: no cover
+
```

Comment:
the ABC client now defines these abstract methods, which need to be implemented by any real client.